rbac/management/group/view.py

```diff
@@ -348,37 +348,21 @@ def update(self, request, *args, **kwargs):
 
     def add_principals(self, group, principals, account=None, org_id=None):
         """Process list of principals and add them to the group."""
-        tenant = self.request.tenant
-
         users = [principal.get("username") for principal in principals]
         if settings.AUTHENTICATE_WITH_ORG_ID:
             resp = self.proxy.request_filtered_principals(users, org_id=org_id, limit=len(users))
-            if "errors" in resp:
-                return resp
-            for item in resp.get("data", []):
-                username = item["username"]
-                try:
-                    principal = Principal.objects.get(username__iexact=username, tenant=tenant)
-                except Principal.DoesNotExist:
-                    principal = Principal.objects.create(username=username, tenant=tenant)
-                    logger.info("Created new principal %s for org_id %s.", username, org_id)
-                group.principals.add(principal)
-                group_principal_change_notification_handler(self.request.user, group, username, "added")
-            return group
         else:
             resp = self.proxy.request_filtered_principals(users, account=account, limit=len(users))
-            if "errors" in resp:
-                return resp
-            for item in resp.get("data", []):
-                username = item["username"]
-                try:
-                    principal = Principal.objects.get(username__iexact=username, tenant=tenant)
-                except Principal.DoesNotExist:
-                    principal = Principal.objects.create(username=username, tenant=tenant)
-                    logger.info("Created new principal %s for account_id %s.", username, account)
-                group.principals.add(principal)
-                group_principal_change_notification_handler(self.request.user, group, username, "added")
-            return group
+
+        if "errors" in resp:
+            return resp
+
```

Comment:
@petracihalova maybe here we can put some code like this below to result **404** if no users were found from BOP:
>
>        if len(resp.get("data", [])) == 0:
>            return {
>                "status_code": status.HTTP_404_NOT_FOUND,
>                "errors": [{"detail": "Users not found.", "status": "404", "source": "principals"}]
>            } 
>